rxandroidble/src/main/java/com/polidea/rxandroidble/RxBleConnection.java

```diff
@@ -417,4 +423,23 @@ public String toString() {
      */
     @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
     Observable<Integer> requestMtu(int mtu);
+
+    /**
+     * <b>This method requires deep knowledge of RxAndroidBLE internals. Use it only as a last resort if you know
+     * what your are doing.</b>
+     * <p>
+     * Queue an operation for future execution. The method accepts a builder parameter. The builder function
+     * will be invoked with three arguments and should return the a non-null instance of {@link RxBleRadioOperation}.
+     * The three arguments the builder receives are:
+     * <ol>
+     * <li>{@link Scheduler} The RxBleRadio scheduler used to execute operation ({@link AndroidSchedulers#mainThread()}</li>
+     * <li>{@link BluetoothGatt} The Android API GATT instance</li>
+     * <li>{@link RxBleGattCallback} The internal Rx ready bluetooth gatt callback to be notified of GATT operations</li>
+     * </ol>
+     *
+     * @param builder The builder used to create the {@link RxBleRadioOperation} instance
+     * @param <T>     The type returned by the {@link RxBleRadioOperation} instance
+     * @return Observable emitting the value after execution or an error in case of failure.
+     */
+    <T> Observable<T> queue(Func3<Scheduler, BluetoothGatt, RxBleGattCallback, RxBleRadioOperation<T>> builder);
```

Comment:
Agree - better to use an interface instead.
Best would be to have an interface with a single entry point as you have proposed because it could be lambdafied although then the custom operation would not be able to specify the `Priority`. A default implementation of a method from Java 8 would be a solution here but we need something different.